src/Components/IndexScene/ShareButtonScene.tsx

```diff
@@ -0,0 +1,264 @@
+                      const url = option.getUrl();
+                      model.onCopyLink(option.shorten, option.absTime, url);
+                      model.setState({
+                        lastSelected: option,
+                      });
+                    }}
+                  />
+                );
+              })}
+            </MenuGroup>
+          );
+        })}
+      </Menu>
+    );
+  };
+
+  static Component = ({ model }: SceneComponentProps<ShareButtonScene>) => {
+    const { lastSelected, isOpen } = model.useState();
+
+    return (
+      <ButtonGroup>
+        <ToolbarButton
+          tooltip={lastSelected.label}
+          icon={lastSelected.icon}
+          variant={'canvas'}
+          narrow={true}
+          onClick={() => {
+            const url = lastSelected.getUrl();
+            model.onCopyLink(lastSelected.shorten, lastSelected.absTime, url);
+          }}
+          aria-label={'Copy shortened URL'}
+        >
+          <span>Share</span>
+        </ToolbarButton>
+        <Dropdown
+          overlay={<ShareButtonScene.MenuActions model={model} />}
+          placement="bottom-end"
+          onVisibleChange={model.setIsOpen.bind(model)}
+        >
+          <ToolbarButton narrow={true} variant={'canvas'} isOpen={isOpen} aria-label={'Open copy link options'} />
+        </Dropdown>
+      </ButtonGroup>
+    );
+  };
+}
+
+const defaultMode: ShortLinkMenuItemData = {
+  key: 'copy-link',
+  label: 'Copy shortened URL',
+  icon: 'share-alt',
+  getUrl: () => undefined,
+  shorten: true,
+  absTime: false,
+};
+
+// Adapted from grafana/grafana/public/app/core/utils/shortLinks.ts shortLinks.ts
+function buildHostUrl() {
+  return `${window.location.protocol}//${window.location.host}${config.appSubUrl}`;
+}
+
+function getRelativeURLPath(url: string) {
+  let path = url.replace(buildHostUrl(), '');
+  return path.startsWith('/') ? path.substring(1, path.length) : path;
+}
+
+export const createShortLink = async function (path: string) {
+  const appEvents = getAppEvents();
+  try {
+    const shortLink = await getBackendSrv().post(`/api/short-urls`, {
+      path: getRelativeURLPath(path),
+    });
+    return shortLink.url;
+  } catch (err) {
+    console.error('Error when creating shortened link: ', err);
+
+    appEvents.publish({
+      type: AppEvents.alertError.name,
+      payload: ['Error generating shortened link'],
+    });
+  }
+};
+
+export const createAndCopyShortLink = async (path: string) => {
+  const appEvents = getAppEvents();
+  const shortLink = await createShortLink(path);
+  if (shortLink) {
+    copyStringToClipboard(shortLink);
+    appEvents.publish({
+      type: AppEvents.alertSuccess.name,
+      payload: ['Shortened link copied to clipboard'],
+    });
+  } else {
+    appEvents.publish({
+      type: AppEvents.alertError.name,
+      payload: ['Error generating shortened link'],
+    });
+  }
+};
+
+export const copyStringToClipboard = (string: string) => {
```

Comment:
There's https://github.com/grafana/explore-logs/blob/main/src/services/text.ts#L5

Reply:
Any reason not to use the existing function?

Reply:
Oh snap I never sent my comment:
copyText seems to require a button to append the textarea for fallback when clipboard.writeText isn't available, we should probably refactor `copyText` to work like `copyStringToClipboard` instead of the other way around. 

Created https://github.com/grafana/explore-logs/issues/986 to track, I can clean this up in the morning

Reply:
No worries. IIRC the other function has a backwards compatibility feature that makes it a bit more resilient.